src/main/java/io/confluent/idesidecar/restapi/credentials/BasicCredentials.java

```diff
@@ -0,0 +1,144 @@
+package io.confluent.idesidecar.restapi.credentials;
+
+import static io.vertx.core.http.HttpHeaders.AUTHORIZATION;
+
+import io.confluent.idesidecar.restapi.exceptions.Failure.Error;
+import io.confluent.idesidecar.restapi.models.ConnectionSpec.SecurityProtocol;
+import io.confluent.idesidecar.restapi.models.ConnectionSpec.SslIdentificationAlgorithm;
+import io.quarkus.runtime.annotations.RegisterForReflection;
+import io.vertx.core.MultiMap;
+import jakarta.validation.constraints.NotNull;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+@Schema(description = "Basic authentication credentials")
+@RegisterForReflection
+public record BasicCredentials(
+    @Schema(
+        description = "The username to use when connecting to the external service.",
+        maxLength = USERNAME_MAX_LEN,
+        minLength = 1
+    )
+    @NotNull
+    String username,
+
+    @Schema(
+        description = "The password to use when connecting to the external service.",
+        maxLength = Password.MAX_LENGTH,
+        minLength = 1
+    )
+    @NotNull
+    Password password
+) implements Credentials {
+
+  private static final int USERNAME_MAX_LEN = 64;
+
+  private static final String PLAIN_LOGIN_MODULE_CLASS =
+      "org.apache.kafka.common.security.plain.PlainLoginModule";
+
+  @Override
+  public Type type() {
+    return Type.BASIC;
+}
+
+  @Override
+  public Optional<Map<String, String>> kafkaClientProperties(
+      KafkaConnectionOptions options
+  ) {
+    var config = new LinkedHashMap<String, String>();
+    var protocol = options.securityProtocol();
+    if (protocol != null && protocol != SecurityProtocol.UNKNOWN) {
+      config.put("security.protocol", protocol.toString());
+    }
```

Comment:
I originally used that approach by defining a boolean `ssl` property instead of `security.protocol`, but I wasn't sure whether we wanted to guarantee that the type of credential and `ssl: true|false` would always determine the `security.protocol`. It does for Basic and API Key+Secret credentials, and will for mTLS, Kerberos and OAuth (and LDAP, which is just Basic). 

So I'm fine with giving that a try -- if that ever changes in the future, the extension may have to migrate any stored connection resources.